src/cli-constant.js

```diff
@@ -1,64 +1,290 @@
+    forwardToApi: true,
+    exception: value => typeof value === "string", // allow path to a parser module
+    choices: ["flow", "babylon", "typescript", "postcss", "json", "graphql"],
+    description: "Specify which parse to use. Defaults to babylon.",
+    getter: (value, argv) => (argv["flow-parser"] ? "flow" : value)
+  },
+  "print-width": {
+    type: "int",
+    category: "format",
+    forwardToApi: true,
+    description:
+      "Specify the length of line that the printer will wrap on. Defaults to 80."
+  },
+  "range-end": {
+    type: "int",
+    category: "format",
+    forwardToApi: true,
+    exception: Infinity,
+    description: dedent(`
+      Format code ending at a given character offset (exclusive).
+      The range will extend forwards to the end of the selected statement.
+      This option cannot be used with --cursor-offset.
+      Defaults to Infinity.
+    `)
+  },
+  "range-start": {
+    type: "int",
+    category: "format",
+    forwardToApi: true,
+    description: dedent(`
+      Format code starting at a given character offset.
+      The range will extend backwards to the start of the first line containing the selected statement.
+      This option cannot be used with --cursor-offset.
+      Defaults to 0.
+    `)
+  },
+  semi: {
+    type: "boolean",
+    category: "format",
+    forwardToApi: true,
+    hidden: true
+  },
+  "single-quote": {
+    type: "boolean",
+    category: "format",
+    forwardToApi: true,
+    description: "Use single quotes instead of double quotes."
+  },
+  stdin: {
+    type: "boolean",
+    description: "Read input from stdin."
+  },
+  "stdin-filepath": {
+    type: "path",
+    forwardToApi: "filepath",
+    description: "Path to the file used to read from stdin."
+  },
+  "tab-width": {
+    type: "int",
+    category: "format",
+    forwardToApi: true,
+    description:
+      "Specify the number of spaces per indentation-level. Defaults to 2."
+  },
+  "trailing-comma": {
+    type: "choice",
+    category: "format",
+    forwardToApi: true,
+    choices: [
+      "none",
+      "es5",
+      "all",
+      { value: "", deprecated: true, redirect: "es5" }
+    ],
+    description:
+      "Print trailing commas wherever possible when multi-line. Defaults to none."
+  },
+  "use-tabs": {
+    type: "boolean",
+    category: "format",
+    forwardToApi: true,
+    description: "Indent lines with tabs instead of spaces."
+  },
+  version: {
+    type: "boolean",
+    category: "command",
+    alias: "v",
+    description: "Print Prettier version."
+  },
+  "with-node-modules": {
+    type: "boolean",
+    category: "config",
+    description: "Process files inside 'node_modules' directory."
+  },
+  write: {
+    type: "boolean",
+    category: "command",
+    description: "Edit the file in-place. (Beware!)"
+  }
+});
+
+const minimistOptions = {
+  boolean: detailOptions
+    .filter(
+      option => !option.name.startsWith("no-") && option.type === "boolean"
```

Comment:
Shouldn’t `--no-semi` and friends be added without the `no-` prefix here, or am I missing something?

Reply:
Both `no-option` and `option` are existed, so I just filter out the `no-` option since `minimist` will handle the `no-` prefix and I'm not sure what will it be if both of them are passed in.

Reply:
Yeah, I wanted to have `'semi'` but not `'no-semi'` in the `boolean` array, but I don’t understand how `'semi'` gets added?

Reply:
`semi` is at L198. 😅 

```js
semi: {
  type: "boolean",
  category: "format",
  forwardToApi: true,
  hidden: true
},
```

Reply:
Oh, I didn’t realize that one has to specify both `foo` and `no-foo` in there! Sounds like it will be super easy to make a mistake there when adding new boolean options in the future?

Reply:
Currently, the `no-option`s are used for docs only, should we put them in their original option with `noDescription`?